cloud/scope/machine.go

```diff
@@ -69,3 +74,35 @@ func NewMachineScope(apiKey string, params MachineScopeParams) (*MachineScope, e
 		LinodeMachine: params.LinodeMachine,
 	}, nil
 }
+
+// GetBootstrapData returns the bootstrap data from the secret in the Machine's bootstrap.dataSecretName.
+func (m *MachineScope) GetBootstrapData(ctx context.Context) (string, error) {
+	if m.Machine.Spec.Bootstrap.DataSecretName == nil {
+		return "", fmt.Errorf(
+			"bootstrap data secret is nil for LinodeMachine %s/%s",
+			m.LinodeMachine.Namespace,
+			m.LinodeMachine.Name,
+		)
+	}
+
+	secret := &corev1.Secret{}
+	key := types.NamespacedName{Namespace: m.LinodeMachine.Namespace, Name: *m.Machine.Spec.Bootstrap.DataSecretName}
+	if err := m.client.Get(ctx, key, secret); err != nil {
+		return "", fmt.Errorf(
+			"failed to retrieve bootstrap data secret for LinodeMachine %s/%s",
+			m.LinodeMachine.Namespace,
+			m.LinodeMachine.Name,
+		)
+	}
+
+	value, ok := secret.Data["value"]
+	if !ok {
+		return "", fmt.Errorf(
+			"bootstrap data secret value key is missing for LinodeMachine %s/%s",
+			m.LinodeMachine.Namespace,
+			m.LinodeMachine.Name,
+		)
+	}
+
+	return b64.StdEncoding.EncodeToString(value), nil
```

Comment:
should the b64 encoding happen by the callee incase anyone is looking for the string? this feels like a generic function and how it's used shouldn't be determined by this func